test/e2e/operator_test.go

```diff
@@ -228,3 +252,195 @@ func setImage(deployment *appsv1.Deployment, image string) {
+			return false, nil
+		}
+		for _, cond := range upstreamResolver.Status.Conditions {
+			if cond.Type == corev1.ContainersReady &&
+				cond.Status == corev1.ConditionTrue {
+				return true, nil
+			}
+		}
+		return false, nil
+	})
+	if err != nil {
+		t.Fatalf("failed to observe ContainersReady condition for pod %s/%s: %v", upstreamResolver.Namespace, upstreamResolver.Name, err)
+	}
+
+	// Create the upstream resolver Service and get the ClusterIP.
+	upstreamSvc := upstreamService(upstreamPodName, upstreamPodNS)
+	if err := cl.Create(context.TODO(), upstreamSvc); err != nil {
+		t.Fatalf("failed to create service %s/%s: %v", upstreamSvc.Namespace, upstreamSvc.Name, err)
+	}
+	defer func() {
+		if err := cl.Delete(context.TODO(), upstreamSvc); err != nil {
+			t.Fatalf("failed to delete service %s/%s: %v", upstreamSvc.Namespace, upstreamSvc.Name, err)
+		}
+	}()
+	if err := cl.Get(context.TODO(), types.NamespacedName{upstreamSvc.Namespace, upstreamSvc.Name}, upstreamSvc); err != nil {
+		t.Fatalf("failed to get service %s/%s: %v", upstreamSvc.Namespace, upstreamSvc.Name, err)
+	}
+	upstreamIP := upstreamSvc.Spec.ClusterIP
+
+	// Update cluster DNS forwarding with the upstream resolver's Service IP address.
+	defaultDNS := &operatorv1.DNS{}
+	if err := cl.Get(context.TODO(), types.NamespacedName{Name: "default"}, defaultDNS); err != nil {
+		t.Fatalf("failed to get default dns: %v", err)
+	}
+	upstream := operatorv1.Server{
+		Name:  "test",
+		Zones: []string{"foo.com"},
+		ForwardPlugin: operatorv1.ForwardPlugin{
+			Upstreams: []string{upstreamIP},
+		},
+	}
+	defaultDNS.Spec.Servers = []operatorv1.Server{upstream}
+	if err := cl.Update(context.TODO(), defaultDNS); err != nil {
+		t.Fatalf("failed to update dns %s: %v", defaultDNS.Name, err)
+	}
+	defer func() {
+		defaultDNS = &operatorv1.DNS{}
+		if err := cl.Get(context.TODO(), types.NamespacedName{Name: "default"}, defaultDNS); err != nil {
+			t.Fatalf("failed to get default dns: %v", err)
+		}
+		if len(defaultDNS.Spec.Servers) != 0 {
+			defaultDNS.Spec = operatorv1.DNSSpec{}
+			if err := cl.Update(context.TODO(), defaultDNS); err != nil {
+				t.Fatalf("failed to update dns %s: %v", defaultDNS.Name, err)
+			}
+		}
+	}()
+
+	// Verify that the Corefile of DNS DaemonSet pods have been updated.
+	dnsDaemonSet := &appsv1.DaemonSet{}
+	if err := cl.Get(context.TODO(), types.NamespacedName{"openshift-dns", "dns-default"}, dnsDaemonSet); err != nil {
+		t.Fatalf("failed to get daemonset %s/%s: %v", dnsDaemonSet.Namespace, dnsDaemonSet.Name, err)
+	}
+	selector, err := metav1.LabelSelectorAsSelector(dnsDaemonSet.Spec.Selector)
+	if err != nil {
+		t.Fatalf("daemonset %s/%s has invalid spec.selector: %v", dnsDaemonSet.Namespace, dnsDaemonSet.Name, err)
+	}
+	defaultDNSPods := &corev1.PodList{}
+	if err := cl.List(context.TODO(), defaultDNSPods, client.MatchingLabelsSelector{Selector: selector}); err != nil {
```

Comment:
Is this selecting all pods that match the selector, in any namespace?  Maybe add a `client.InNamespace(dnsDaemonSet.Namespace)` argument.

Reply:
ack